src/uu/printf/src/printf.rs

```diff
@@ -30,12 +33,33 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
     let format_string = matches
         .get_one::<String>(options::FORMATSTRING)
         .ok_or_else(|| UUsageError::new(1, "missing operand"))?;
-    let values: Vec<String> = match matches.get_many::<String>(options::ARGUMENT) {
-        Some(s) => s.map(|s| s.to_string()).collect(),
+
+    let values: Vec<_> = match matches.get_many::<String>(options::ARGUMENT) {
+        Some(s) => s.map(|s| FormatArgument::Unparsed(s.to_string())).collect(),
         None => vec![],
     };
 
-    printf(format_string, &values[..])?;
+    let mut args = values.iter().peekable();
+    for item in parse_spec_and_escape(format_string.as_ref()) {
+        match item?.write(stdout(), &mut args)? {
```

Comment:
Why do you use `item?` here but a match on `item` on line 52?

Reply:
Yeah that is curious hahaha. I seem to have changed line 52 last, but if I change it back no test is failing and I think the question mark is the correct behaviour.